### PR TITLE
feat(projects.projet): Add Antifraud check for project discovery activation banner

### PR DESCRIPTION
**Pull request by brofik**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-discover` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-543 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default class PciProjectDashboardActivateBannerController {
+  $onInit() {
+    console.log('PciProjectDashboardActivateBannerController: this: ', this);
+  }
+}
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js:5]:
```suggestion
```
No needed or if it's for anticipate the future you can put a message "todo" for example

File: packages/manager/modules/pci/src/projects/project/project.routing.js [hunk 2]
```diff
@@ -56,6 +56,10 @@ export default /* @ngInject */ ($stateProvider) => {
       activateDiscovery: /* @ngInject */ ($transition$, isDiscoveryProject) =>
         isDiscoveryProject && $transition$.params().activateDiscovery,
 
+      isManuallyReviewedByAntiFraud: /* @ngInject */ (PciProject) => {
+        return PciProject.isManuallyReviewedByAntiFraud();
+      },
+
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/project.routing.js:62]:
```suggestion
      isManuallyReviewedByAntiFraud: /* @ngInject */ (PciProject) => PciProject.isManuallyReviewedByAntiFraud(),
```

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default class PciProjectDashboardActivateBannerController {
+  $onInit() {
+    console.log('PciProjectDashboardActivateBannerController: this: ', this);
```
Comment (dectotam) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js:3]:
```suggestion
```

File: packages/manager/modules/pci/src/projects/project/project.service.js [hunk 4]
```diff
@@ -16,6 +20,43 @@ export default class {
...
+      })
+      .catch(() => {
+        return false;
+      });
+  }
+
+  getOrderFollowUp(orderId) {
+    const endPoint = `/me/order/${orderId}/followUp`;
+    return this.$http.get(endPoint).then(({ data }) => data);
+  }
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/project.service.js:58]:
```suggestion
  getOrderFollowUp(orderId) {
    return this.$http.get(`/me/order/${orderId}/followUp`).then(({ data }) => data);
  }
```

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.product-activate-banner {
+  top: -1.47rem !important;
+}
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss:3]: Are you sure that is used ?

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                    data-translate="pci_projects_project_discovery_activate_banner_description"
+                ></span>
+            </div>
+        </div>
+        <oui-button
+            class="d-block d-sm-none"
+            size="l"
+            block
+            variant="primary"
+            on-click="$ctrl.onClick()"
```
Comment (dectotam) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html:45]:
```suggestion
            data-on-click="$ctrl.onClick()"
```

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                <span
+                    data-translate="pci_projects_project_discovery_activate_banner_description"
+                ></span>
+            </div>
+        </div>
+        <oui-button
+            class="d-block d-sm-none"
+            size="l"
+            block
+            variant="primary"
```
Comment (dectotam) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html:44]:
```suggestion
            data-variant="primary"
```

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                >
+                    <span
+                        data-translate="pci_projects_project_discovery_activate_banner_no_promotion_title"
+                    ></span>
+                </h6>
+                <span
+                    data-translate="pci_projects_project_discovery_activate_banner_description"
+                ></span>
+            </div>
+            <oui-button variant="primary" on-click="$ctrl.onClick()">
```
Comment (dectotam) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html:83]:
```suggestion
            <oui-button data-variant="primary" data-on-click="$ctrl.onClick()">
```

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js (hunk 1 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js]: the controller is removed.

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss (hunk 5 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss:3]: This CSS class is used in the template packages/manager/modules/pci/src/projects/project/components/activate-project-banner/activate-project-banner.html
And is already loaded in packages/manager/modules/pci/src/projects/project/components/activate-project-banner/index.js

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js (hunk 3 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.controller.js]: same as https://github.com/ovh/manager/pull/10789#discussion_r1467743475

**Review by ghyenne — APPROVED** [review]

**Review by dectotam — APPROVED** [review]

**Review by y4nnL — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/project-dashboard-activate-banner.html:1]: question: I don't see `discoveryPromotionVoucherAmount` anywhere in this component, don't you want to use `voucherAmount` instead ?

File: packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/components/project-dashboard-activate-banner/index.scss:1]: Rename to `project-dashboard-activate-banner.scss`

File: packages/manager/modules/pci/src/projects/project/project.service.js [hunk 9]
```diff
@@ -16,6 +20,42 @@ export default class {
...
+        const validatingOrder = orders.find(
+          (order) =>
+            order.status === ORDER_FOLLOW_UP_STEP_ENUM.DELIVERING.toLowerCase(),
+        );
+
+        return validatingOrder
+          ? this.getOrderFollowUp(validatingOrder?.orderId)
+          : false;
+      })
+      .then((data) => {
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/project.service.js:37]:
```suggestion
      .then((orderFollowUp) => {
```

**Review by y4nnL — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/project.service.js [hunk 10]
```diff
@@ -16,6 +20,46 @@ export default class {
...
+                ORDER_FOLLOW_UP_HISTORY_STATUS_ENUM.FRAUD_MANUAL_REVIEW,
+            ) || false
+          );
+        }
+        return false;
+      })
+      .catch(() => {
+        return false;
+      });
+  }
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/project.service.js:55]: The complexity of this method could be reduced
```suggestion
  isManuallyReviewedByAntiFraud() {
    return this.$http
      .get(`/cloud/order/?planCode=project.2018`)
      .then(({ data: orders }) => {
        const { DELIVERING, VALIDATING } = ORDER_FOLLOW_UP_STEP_ENUM;
        const { FRAUD_MANUAL_REVIEW } = ORDER_FOLLOW_UP_HISTORY_STATUS_ENUM;
        const { orderId } = orders.find(
          ({ status }) => status.toLowerCase() === DELIVERING.toLowerCase(),
        );
        return orderId
          ? this.getOrderFollowUp(orderId).then((orderFollowUp) =>
              orderFollowUp
                .find(({ step }) => step === VALIDATING)
                .history.some(({ label }) => label === FRAUD_MANUAL_REVIEW),
            )
          : false;
      })
      .catch(() => false);
  }
```

**Review by y4nnL — CHANGES_REQUESTED** [review]

**Review by y4nnL — APPROVED** [review]

**Pull request merged**